tiledb/libtiledb.pyx

```diff
@@ -2551,6 +2548,9 @@ cdef class Dim(object):
             # coordinates are written.
             dim_datatype = TILEDB_STRING_ASCII
         else:
+            if domain is None or len(domain) != 2:
```

Comment:
We need to handle `str` / `"U*"` dtypes as well - this only works for `bytes`.

Reply:
(also IMO `domain=(None,None)` is still _valid_ just not required, so let's keep the `len` check too)

Reply:
OK I see.